assets/src/components/editing/toolbar/HoverContainer.tsx

```diff
@@ -32,13 +32,18 @@ export const HoverContainer = (props: PropsWithChildren<Props>) => {
     [],
   );
 
-  const children = <span style={{ ...props.style }}>{props.children}</span>;
-
-  if (!isOpen) return children;
```

Comment:
This early return ended up being the problem.

If the structure of the react-tree changes, then slate does some weird things. The solution was to always render the `<Popover>` so the hierarchy doesn't change, but selectively tell it to be open or not.